test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +94,7 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (check.isMutating()) assertEquals(addressBook, saveFile.load());
```

Comment:
it would also be good to check if the command that is NOT mutating indeed doesn't save the file